crates/noirc_frontend/src/hir/type_check/stmt.rs

```diff
@@ -287,13 +288,34 @@ impl<'interner> TypeChecker<'interner> {
                     if v >= max {
                         self.errors.push(TypeCheckError::OverflowingAssignment {
                             expr: value,
-                            ty: annotated_type.clone(),
+                            typ: annotated_type.clone(),
                             range: format!("0..={}", max - 1),
                             span,
                         });
                     };
                 };
             }
+            HirExpression::Infix(HirInfixExpression { operator, rhs, .. }) => {
+                if operator.is_bit_shift() {
+                    let rhs_expr = self.interner.expression(&rhs);
+                    let rhs_value: u128 = match rhs_expr {
+                        HirExpression::Literal(HirLiteral::Integer(value)) => value.to_u128(),
+                        _ => return,
+                    };
+                    if let Type::Integer(_, bit_count) = annotated_type {
+                        let max: u128 = 1 << bit_count;
```

Comment:
This needs to be done in SSA or later. The type checker cannot catch cases like `let z: u8 = 3 >> (4 + 4);`